src/applications/travel-pay/containers/TravelPayStatusApp.jsx

```diff
@@ -69,24 +88,40 @@ export default function App({ children }) {
   }
 
   const resetSearch = () => {
-    setAppliedStatuses(statusesToFilterBy);
-    setCheckedStatuses(statusesToFilterBy);
+    setAppliedStatusFilters([]);
+    setCheckedStatusFilters([]);
+    setAppliedDateFilter('all');
+    setSelectedDateFilter('all');
+    setCurrentPage(1);
   };
 
   const applyFilters = () => {
-    setAppliedStatuses(checkedStatuses);
+    setAppliedStatusFilters(checkedStatusFilters);
+    setAppliedDateFilter(selectedDateFilter);
   };
 
   const onStatusFilterChange = (e, statusName) => {
     if (e.currentTarget.checked) {
-      setCheckedStatuses([...checkedStatuses, statusName]);
+      setCheckedStatusFilters([...checkedStatusFilters, statusName]);
     } else {
-      setCheckedStatuses(
-        checkedStatuses.filter(statusFilter => statusFilter !== statusName),
+      setCheckedStatusFilters(
+        checkedStatusFilters.filter(
+          statusFilter => statusFilter !== statusName,
+        ),
       );
     }
   };
 
+  const onDateFilterChange = e => {
+    setSelectedDateFilter(e.target.value);
+    setCurrentPage(1);
```

Comment:
this sets the current page even when the filters haven't been applied. Shouldn't we only reset the current page after a search is submitted?

Reply:
Good catch, meant for that to be in `applyFilters`, not `onDateFilterChange`